shared/src/main/scala/io/lemonlabs/uri/Uri.scala

```diff
@@ -1086,6 +1140,23 @@ final case class DataUrl(mediaType: MediaType, base64: Boolean, data: Array[Byte
 
   private[uri] def toString(c: UriConfig): String =
     scheme + ":" + pathString(c)
+
+  override def equals(obj: Any): Boolean = obj match {
```

Comment:
There was a bug here. `equals` and `hashCode` were not working at all for `DataUrl` since it has a `Array[Byte]` which does equality by reference rather than value...

Added manual implementations to fix this